pkg/database/document_database.go

```diff
@@ -0,0 +1,193 @@
+	GetDocument(ctx context.Context, req *schemav2.DocumentSearchRequest) (*schemav2.DocumentSearchResponse, error)
+	// CreateDocument creates a new document
+	CreateDocument(ctx context.Context, req *schemav2.DocumentInsertRequest) (*schemav2.DocumentInsertResponse, error)
+	// DocumentAudit returns the document audit history
+	DocumentAudit(ctx context.Context, req *schemav2.DocumentAuditRequest) (*schemav2.DocumentAuditResponse, error)
+}
+
+func (d *db) ListCollections(ctx context.Context, req *schemav2.CollectionListRequest) (*schemav2.CollectionListResponse, error) {
+	collections, err := d.documentEngine.ListCollections(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	cinfos := make([]*schemav2.CollectionInformation, 0, len(collections))
+	for collectionName, indexes := range collections {
+		cinfos = append(cinfos, newCollectionInformation(collectionName, indexes))
+	}
+
+	return &schemav2.CollectionListResponse{Collections: cinfos}, nil
+}
+
+func (d *db) getCollection(ctx context.Context, collectionName string) (*schemav2.CollectionInformation, error) {
+	indexes, err := d.documentEngine.GetCollection(ctx, collectionName)
+	if err != nil {
+		return nil, err
+	}
+
+	return newCollectionInformation(collectionName, indexes), nil
+}
+
+// GetCollection returns the collection schema
+func (d *db) GetCollection(ctx context.Context, req *schemav2.CollectionGetRequest) (*schemav2.CollectionGetResponse, error) {
+	cinfo, err := d.getCollection(ctx, req.Name)
+	if err != nil {
+		return nil, err
+	}
+
+	return &schemav2.CollectionGetResponse{Collection: cinfo}, nil
+}
+
+// CreateCollection creates a new collection
+func (d *db) CreateCollection(ctx context.Context, req *schemav2.CollectionCreateRequest) (*schemav2.CollectionCreateResponse, error) {
+	indexKeys := make(map[string]sql.SQLValueType)
+
+	// validate index keys
+	for name, pk := range req.IndexKeys {
+		schType, isValid := schemaToValueType[pk.Type]
+		if !isValid {
+			return nil, fmt.Errorf("invalid index key type: %v", pk)
+		}
+		// TODO: add support for other types
+		// TODO: add support for auto increment
+		indexKeys[name] = schType
+	}
+
+	err := d.documentEngine.CreateCollection(ctx, req.Name, indexKeys)
+	if err != nil {
+		return nil, err
+	}
+
+	// get collection information
+	cinfo, err := d.getCollection(ctx, req.Name)
+	if err != nil {
+		return nil, err
+	}
+
+	return &schemav2.CollectionCreateResponse{Collection: cinfo}, nil
+}
+
+// CreateDocument creates a new document
+func (d *db) CreateDocument(ctx context.Context, req *schemav2.DocumentInsertRequest) (*schemav2.DocumentInsertResponse, error) {
+	docID, err := d.documentEngine.CreateDocument(ctx, req.Collection, req.Document)
+	if err != nil {
+		return nil, err
+	}
+
+	return &schemav2.DocumentInsertResponse{DocumentId: docID.Hex()}, nil
+}
+
+// GetDocument returns the document
+func (d *db) GetDocument(ctx context.Context, req *schemav2.DocumentSearchRequest) (*schemav2.DocumentSearchResponse, error) {
```

Comment:
if multiple documents may be returned then it might be better to call it `GetDocuments`